internal/pipe/linkedin/client.go

```diff
@@ -0,0 +1,134 @@
+package linkedin
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/goreleaser/goreleaser/pkg/context"
+	"golang.org/x/oauth2"
+)
+
+type OAuthClientConfig struct {
+	Context     *context.Context
+	AccessToken string
+}
+
+type Client struct {
+	client  *http.Client
+	baseURL string
+}
+
+type PostShareText struct {
+	Text string `json:"text"`
+}
+
+type PostShareRequest struct {
+	Text  PostShareText `json:"text"`
+	Owner string        `json:"owner"`
+}
+
+func CreateLinkedInClient(cfg OAuthClientConfig) (Client, error) {
```

Comment:
since we are using this only inside this pkg, I think all those structs/funcs can be unexported...